.github/actions/build/build-strimzi-binaries/action.yml

```diff
@@ -109,6 +109,15 @@ runs:
         name: strimzi-binaries.tar
         path: strimzi-binaries.tar
 
+    # Clean external SNAPSHOT dependencies before caching
+    - name: Clean external SNAPSHOT dependencies from Maven repository
+      if: github.ref == 'refs/heads/main'
```

Comment:
Apologies for dropping the comments like this one by one. 🤦  Why do we do this only in main branch? Given the primary issue is non-Strimzi SNAPSHOT dependencies, does the Strimzi branch matter?